lib/internal/modules/cjs/loader.js

```diff
@@ -84,6 +84,25 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  const stack = err.stack.split('\n');
+
+  const lineWithErr = stack[1];
+
+  /*
+    The regular expression below targets the most common import statement
+    usage. However, some cases are not matching, cases like import statement
+    after a comment block and/or after a variable definition.
+  */
+  if (err.message.startsWith('Unexpected token export') ||
+    (/^\s*import(?=[ {'"*])\s*(?![ (])/).test(lineWithErr)) {
+    err.stack = 'To load an ES module, set "type": "module" ' +
```

Comment:
Shouldn't text like this be added to `err.message` or something like that? Wedging stuff that isn't a stack trace into `err.stack` seems like the wrong place to put it, no?

Reply:
actually yeah, good call - this will make the stacks violate the spec, if i can ever get that proposal advanced.

Reply:
I believe the reasoning here was originally that if we append the _message_ the error looks like:

```
(node:7720) ExperimentalWarning: The ESM module loader is experimental.

C:\Users\Guy\x.js:1
export var p = 5;
^^^^^^

To load an ES module using --experimental-modules set "type": "module" in
the package.json or use the .mjs extension.

SyntaxError: Unexpected token export
    at Module._compile (internal/modules/cjs/loader.js:720:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at internal/modules/esm/translators.js:87:15
    at Object.meta.done (internal/modules/esm/create_dynamic_module.js:48:9)
    at file:///C:/Users/Guy/x.js:9:13
    at ModuleJob.run (internal/modules/esm/module_job.js:111:37)
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async Loader.import (internal/modules/esm/loader.js:134:24)
```

where the note is hidden beneath the stack trace.

Ideally we'd have more control over mutating these error outputs though.

Reply:
Correction - I mean the note is hidden beneath the error frame.

Reply:
Maybe an acceptable solution would be to leave the error contents alone and emit a separate warning with the note?

Reply:
You could even overwrite the own data property with an accessory that console.warned.

Reply:
> Maybe an acceptable solution would be to leave the error contents alone and emit a separate warning with the note?

So call `console.warn` after the error condition, would be enough, right? it would still print before the error `message` and `stack`.